log4j-api/src/main/java/org/apache/logging/log4j/util/LoaderUtil.java

```diff
@@ -155,9 +155,11 @@ public ClassLoader run() {
 	if (!isForceTccl()) {
             accumulateClassLoaders(LoaderUtil.class.getClassLoader(), classLoaders);
             accumulateClassLoaders(tcl == null ? null : tcl.getParent(), classLoaders);
-            final ClassLoader systemClassLoader = ClassLoader.getSystemClassLoader();
-            if (systemClassLoader != null) {
-                classLoaders.add(systemClassLoader);
+            if (!GET_CLASS_LOADER_DISABLED) {
+                final ClassLoader systemClassLoader = ClassLoader.getSystemClassLoader();
+                if (systemClassLoader != null) {
```

Comment:
For what it's worth, `ClassLoader.getSystemClassLoader()` appears to be guaranteed to not return null. I left it as-is for now.